RockSniffer/RPC/DiscordRPCHandler.cs

```diff
@@ -70,6 +76,13 @@ internal void UpdatePresence()
             //If we have a valid song and are playing a song
             if ((songdetails != null && readout != null) && (state == SnifferState.SONG_STARTING || state == SnifferState.SONG_PLAYING || state == SnifferState.SONG_ENDING))
             {
+                // Get the appropriate album cover
+                if (albumArtResolver != null && albumArtResolver.Get(songdetails) is (string URL, string DisplayText) resultTuple)
```

Comment:
This will still crash, if `albumArtResolver.Get` throws any exception.

Reply:
I'm honestly not sure. I think the intent should be for `Get` to be infallible, returning `null` on error. Would wrapping the entire body of `Get` in `try-catch` be a fine solution?

Reply:
Shouldn't wrapping the `Get` call be enough? I may be too paranoid, but I wouldn't want the app to crash if something unexpected happens when calling the API, even if all cases seem to be handled.

Reply:
No, I get that. I thought the code in `UpdatePresence` is complicated enough already, but now I see it's not that bad.